diff -Naur amdgpu-pro-17.10-401251-org/amd/amdkcl/kcl_drm.c amdgpu-pro-17.10-401251/amd/amdkcl/kcl_drm.c
--- amdgpu-pro-17.10-401251-org/amd/amdkcl/kcl_drm.c	2017-03-22 04:29:19.000000000 +0100
+++ amdgpu-pro-17.10-401251/amd/amdkcl/kcl_drm.c	2017-04-08 12:51:07.496355031 +0200
@@ -386,7 +386,7 @@
 }
 EXPORT_SYMBOL(drm_atomic_helper_disable_all);
 
-#ifndef OS_NAME_RHEL_6
+#if !defined(OS_NAME_RHEL_6) && LINUX_VERSION_CODE < KERNEL_VERSION(4, 4, 0)
 struct drm_atomic_state *
 drm_atomic_helper_duplicate_state(struct drm_device *dev,
 				  struct drm_modeset_acquire_ctx *ctx)
diff -Naur amdgpu-pro-17.10-401251-org/include/kcl/kcl_drm.h amdgpu-pro-17.10-401251/include/kcl/kcl_drm.h
--- amdgpu-pro-17.10-401251-org/include/kcl/kcl_drm.h	2017-03-22 04:29:24.000000000 +0100
+++ amdgpu-pro-17.10-401251/include/kcl/kcl_drm.h	2017-04-08 12:51:07.497355031 +0200
@@ -53,7 +53,7 @@
 			     struct drm_modeset_acquire_ctx *ctx);
 int drm_atomic_helper_disable_all(struct drm_device *dev,
 				  struct drm_modeset_acquire_ctx *ctx);
-#ifndef OS_NAME_RHEL_6
+#if !defined(OS_NAME_RHEL_6) && LINUX_VERSION_CODE < KERNEL_VERSION(4, 4, 0)
 struct drm_atomic_state *
 drm_atomic_helper_duplicate_state(struct drm_device *dev,
 				  struct drm_modeset_acquire_ctx *ctx);
